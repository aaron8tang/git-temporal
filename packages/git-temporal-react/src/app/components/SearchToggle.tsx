import React from 'react';
import { style } from 'app/styles';

// import { SearchInput } from 'app/components/SearchInput';
import { SearchIcon } from 'app/components/SearchIcon';
import { SearchInput } from 'app/components/SearchInput';

export interface SearchToggleProps {
  value: string;
  onChange: (value) => void;
  placeholder?: string;
  style?: object;
}

const initialState = {
  showSearchInput: false,
};

type SearchToggleState = Readonly<typeof initialState>;

const iconStyle = {
  _extends: ['selectable'],
  width: 25,
  height: 23,
  textAlign: 'center',
  paddingTop: 6,
  marginRight: 12,
};

export class SearchToggle extends React.Component<
  SearchToggleProps,
  SearchToggleState
> {
  readonly state: SearchToggleState = initialState;

  render() {
    const { value, onChange } = this.props;
    if (
      (value && value.toString().trim() !== '') ||
      this.state.showSearchInput
    ) {
      return (
        <SearchInput
          value={value}
          onChange={onChange}
          onClear={this.onClose}
          style={style(this.props.style)}
        />
      );
    }
    return (
      <div
        style={style(iconStyle, this.props.style, {
          width: iconStyle.width,
        })}
        onClick={this.onIconClick}
      >
        <SearchIcon width={16} height={16} />
      </div>
    );
  }
  onIconClick = () => {
    this.setState({ showSearchInput: true });
  };

  onClose = () => {
    this.setState({ showSearchInput: false });
    this.props.onChange('');
  };
}
