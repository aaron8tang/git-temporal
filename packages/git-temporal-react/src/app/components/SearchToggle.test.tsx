import React from 'react';
import { shallow } from 'enzyme';

import { SearchToggle } from './SearchToggle';

describe('components/SearchToggle', () => {
  describe('when rendered with value', () => {
    let wrapper;
    beforeAll(() => {
      wrapper = shallow(
        <SearchToggle value={'test value'} onChange={jest.fn()} />
      );
    });

    test('it should match snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });
  });
});
