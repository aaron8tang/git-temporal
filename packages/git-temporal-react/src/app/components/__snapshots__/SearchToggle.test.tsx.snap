// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/SearchToggle when rendered with value it should match snapshot 1`] = `
<Component
  onChange={[MockFunction]}
  onClear={[Function]}
  style={Object {}}
  value="test value"
/>
`;
